--- a/gst/parse/meson.build	Wed Oct 12 17:39:51 2022
+++ b/gst/parse/meson.build	Thu Oct 13 14:49:30 2022
@@ -31,21 +31,9 @@
 bison_cdata = configuration_data()
 
 bison_min_version='2.4'
-bison = find_program('bison', 'win_bison')
+bison = find_program('bison', 'win_bison', 'yacc')
 
-bversion_res = run_command([bison, '--version'], check: true)
-bversion = bversion_res.stdout().split('\n')[0].split(' ')[-1].strip()
-if bversion.version_compare('<' + bison_min_version)
-  error('bison version @0@ >= @1@: NO'.format(bversion, bison_min_version))
-else
-  message('bison version @0@ >= @1@: YES'.format(bversion, bison_min_version))
-endif
-
-if bversion.version_compare('>' + '2.5')
-  bison_parser_cdata.set('BISON_PURE_PARSER', '%define api.pure full')
-else
-  bison_parser_cdata.set('BISON_PURE_PARSER', '%pure-parser')
-endif
+bison_parser_cdata.set('BISON_PURE_PARSER', '%pure-parser')
 
 gen_grammar_file = configure_file(input : 'grammar.y.in',
   output : 'grammar.y',
