--- a/contrib/kiss-chbuild
+++ b/contrib/kiss-chbuild
@@ -15,18 +15,19 @@
 
 cd "${cac_dir:=$KISS_ROOT${XDG_CACHE_HOME:-$HOME/.cache}/kiss}"
 
-[ -f kiss-chroot.tar.xz ] || {
+[ -f carbs-rootfs.tar.xz ] || {
     log "Downloading chroot tarball"
-    wget https://dl.getkiss.org/kiss-chroot.tar.xz
+    wget https://dl.carbslinux.org/releases/carbs-rootfs.tar.xz
 }
 
-[ -d kiss-chroot ] || {
+[ -d carbs-chroot ] || {
     log "Extracting chroot"
-    tar xvf kiss-chroot.tar.xz
+    mkdir -p carbs-chroot
+    tar xvf carbs-rootfs.tar.xz -C carbs-chroot
 }
 
 log "Creating temporary chroot"
-cp -a kiss-chroot "chroot-$pid"
+cp -a carbs-chroot "chroot-$pid"
 
 trap clean EXIT INT
 
