--- python/Lib/ensurepip/__init__.py.orig	Wed Oct  6 02:23:28 2021
+++ python/Lib/ensurepip/__init__.py	Wed Oct  6 02:23:47 2021
@@ -180,7 +180,7 @@
             additional_paths.append(filename)
 
         # Construct the arguments to be passed to the pip command
-        args = ["install", "--no-cache-dir", "--no-index", "--find-links", tmpdir]
+        args = ["install", "-I", "--no-cache-dir", "--no-index", "--find-links", tmpdir]
         if root:
             args += ["--root", root]
         if upgrade:
