--- iwd.orig/src/netconfig.c	Tue Sep 14 23:01:22 2021
+++ iwd/src/netconfig.c	Thu Sep 16 12:38:50 2021
@@ -185,7 +185,7 @@
 	struct in6_addr in6_addr;
 	char *addr_str = l_malloc(INET6_ADDRSTRLEN);
 
-	memcpy(in6_addr.__in6_u.__u6_addr8, addr, 16);
+	memcpy(in6_addr.__in6_union.__s6_addr, addr, 16);
 
 	if (L_WARN_ON(unlikely(!inet_ntop(AF_INET6, &in6_addr, addr_str,
 						INET6_ADDRSTRLEN)))) {
